src/core/internal/hash.d

```diff
@@ -553,7 +565,8 @@ size_t hashOf(T)(scope const T val, size_t seed = 0) if (!is(T == enum) && is(T
 @nogc nothrow pure @trusted
 size_t hashOf(T)(scope const T val)
 if (!is(T == enum) && (is(T == interface) || is(T == class))
-    && isFinalClassWithAddressBasedHash!T)
+    && isFinalClassWithAddressBasedHash!T
+    || isCppClassWithoutHash!T)
```

Comment:
As this is used three times it might make sense to define `canBitwiseHash` as a separate template.